dbms/src/Storages/Page/V3/WAL/serialize.cpp

```diff
@@ -114,13 +153,20 @@ void serializePutTo(const EditRecord & record, WriteBuffer & buf)
     serializeVersionTo(record.version, buf);
     writeIntBinary(record.being_ref_count, buf);
 
-    serializeEntryTo(record.entry, buf);
+    if constexpr (std::is_same_v<EditRecord, u128::PageEntriesEdit::EditRecord>)
+    {
+        serializeEntryTo(record.entry, buf);
+    }
+    else if constexpr (std::is_same_v<EditRecord, universal::PageEntriesEdit::EditRecord>)
+    {
+        serializeEntryWithCheckpointInfoTo(record.entry, buf);
```

Comment:
maybe we can put

- has_checkpoint_info or not
- local data reclaimed

as two bits in the `flags` field?

Reply:
Great suggestion. And I find we don't even need to store the info about whether local data is reclaimed. We can determine it just from `entry.file_id`.